spec/lib/miq_automation_engine/models/miq_ae_git_import_spec.rb

```diff
@@ -0,0 +1,113 @@
+describe MiqAeGitImport do
+  before do
+    EvmSpecHelper.local_guid_miq_server_zone
+    @user = FactoryGirl.create(:user_with_group)
+  end
+
+  context "git import" do
+    let(:branch_name) { "b1" }
+    let(:tag_name) { "t1" }
+    let(:domain_name) { "BB8" }
+    let(:url) { "http://www.example.com/x/y" }
+    let(:domain) do
+      FactoryGirl.create(:miq_ae_git_domain,
+                         :tenant => @user.current_tenant,
+                         :name   => domain_name)
+    end
+    let(:repo) { FactoryGirl.create(:git_repository, :url => url) }
+    let(:branch_hash) do
+      {'ref' => branch_name, 'ref_type' => MiqAeDomain::BRANCH}
+    end
+
+    let(:tag_hash) do
+      {'ref' => tag_name, 'ref_type' => MiqAeDomain::TAG}
+    end
+
+    let(:branch) { FactoryGirl.create(:git_branch, :name => branch_name) }
+    let(:tag) { FactoryGirl.create(:git_tag, :name => tag_name) }
+
+    it "import a domain given a git repository and a branch" do
```

Comment:
I think this needs to be structured a bit differently, so you can leverage a before block to reduce some duplication. Maybe something like this? https://gist.github.com/eclarizio/82aa57e6ca891ef1dc6d3b99f8ccbfec

Overall, though, I think there's too much going on here and it's going to involve a lot more specs with a lot more contexts to make sure everything is covered. I think it would be simpler if even just the `default_import_options` and `validate_ref` methods were separate classes.